File: app/src/main/java/com/slapshotapps/dragonshockey/ViewUtils/interfaces/StickyHeaderAdapter.java
package com.slapshotapps.dragonshockey.ViewUtils.interfaces;

import androidx.recyclerview.widget.RecyclerView;

/**
 * Created by willmetz on 8/6/16.
 */

public interface StickyHeaderAdapter<T extends RecyclerView.ViewHolder> {

    T onCreateHeaderViewHolder(RecyclerView parent);

    void onBindHeaderViewHolder(T viewHolder);
}

File: app/src/main/java/com/slapshotapps/dragonshockey/ViewUtils/interfaces/StickyHeaderAdapter.kt
package com.slapshotapps.dragonshockey.ViewUtils.interfaces

import androidx.recyclerview.widget.RecyclerView
import androidx.viewbinding.ViewBinding

/**
 * Created by willmetz on 8/6/16.
 */
interface StickyHeaderAdapter<T : ViewBinding> {
    fun onCreateViewBinding(parent: RecyclerView): T
    fun onBindHeaderViewHolder(binding: ViewBinding): Unit
}